app/src/test/scala/org/alephium/explorer/persistence/queries/BlockQueriesSpec.scala

```diff
@@ -155,4 +158,42 @@ class BlockQueriesSpec
       }
     }
   }
+
+  "mainChainQuery" should {
+
+    //Build database state for headers and run explain on mainChainQuery
+    def runExplain(headers: Seq[BlockHeader]) = {
+      //Clear table persist test-data
+      run(BlockHeaderSchema.table.delete).futureValue
+      run(BlockHeaderSchema.table ++= headers).futureValue
+
+      //build explain query for mainChainQuery
+      run(BlockQueries.mainChainQuery.result.explainFlatten()).futureValue
+    }
+
+    "use block_headers_main_chain_idx" when {
+      "data size is large" taggedAs IndexCheck in {
+        /*
+         * This test is flaky. Postgres randomly choose not to use `block_headers_main_chain_idx`.
```

Comment:
hum... I'm not fan of adding a flaky test, even tho I understand the all context, this will be too boring to have PR failing because of this. Let's try to see if we really need this `main_chain` in the end.

Reply:
The test seems necessary, even if it's flaky because without this test we wouldn't have known that this problem exists.

Nice. if we can remove the need for `main_chain` all together that'd be great. But are you referring to `main_chain` column or `main_chain` index?

Reply:
Could you try to generate a random dataset with 200 data records and 198 of them are  `main_chain = true`? Let's see if it will flaky still.

Reply:
Low randomness in data skips the index.

Your test:

| main_chain | count(*) |
|------------|-------|
| false      | 2     |
| true       | 198   |

> Seq Scan on block_headers  (cost=0.00..12.80 rows=140 width=260)
  Filter: main_chain

Reply:
Thanks. Could you also paste the two different query analysis results so I could see the exact plans?

Reply:
//EXPLAIN (Under the table above)
Seq Scan on block_headers  (cost=0.00..12.80 rows=140 width=260)
  Filter: main_chain

//EXPLAIN ANALYZE
Seq Scan on block_headers  (cost=0.00..12.80 rows=140 width=260) (actual time=0.018..0.091 rows=198 loops=1)
  Filter: main_chain
  Rows Removed by Filter: 2
Planning Time: 0.796 ms
Execution Time: 0.139 ms

Reply:
Maybe I missed something, but both are using the `main_chain` index right?

Reply:
They both are using `main_chain` column but not the index because of less variation in data.

The query being analysed is [BlockQueries.mainChainQuery](https://github.com/alephium/explorer-backend/blob/master/app/src/main/scala/org/alephium/explorer/persistence/queries/BlockQueries.scala#L46).

We get the same plan output when the query is run directly in Postgres. 

```sql
EXPLAIN ANALYSE
SELECT *
FROM block_headers
WHERE main_chain = true;
```

Reply:
Could you show me the different execution plans (EXPLAIN query)? I want to see the difference and might get some new ideas from the plan outputs.

Reply:
> We get the same plan output when the query is run directly in Postgres.

@simerplaha please paste the different plan outputs in the flaky test (when test again 200 data records with 198 of them have `main_chain = true`). I cannot guess what are the different outputs, and I don't have the time to get them by myself. Thank you in advance!

Reply:
Hey Cheng, yep definitely.

Just wanted to run some additional tests supplementing your request to figure out why this index is misbehaving.

Will surely sort this out. Sorry for the delay.

Reply:
Does [this](https://github.com/alephium/explorer-backend/pull/292#issuecomment-1186783387) report match what you are asking for? The graphs should give a sense of probability.